conan/cli/commands/graph.py

```diff
@@ -298,3 +300,110 @@ def graph_explain(conan_api, parser,  subparser, *args):
+                                               conanfile_path=path,
+                                               cwd=cwd,
+                                               partial=args.lockfile_partial,
+                                               overrides=overrides)
+    profile_host, profile_build = conan_api.profiles.get_profiles_from_args(args)
+
+    if path:
+        deps_graph = conan_api.graph.load_graph_consumer(path, args.name, args.version,
+                                                         args.user, args.channel,
+                                                         profile_host, profile_build, lockfile,
+                                                         remotes, args.update,
+                                                         check_updates=args.check_updates,
+                                                         is_build_require=args.build_require)
+    else:
+        deps_graph = conan_api.graph.load_graph_requires(args.requires, args.tool_requires,
+                                                         profile_host, profile_build, lockfile,
+                                                         remotes, args.update,
+                                                         check_updates=args.check_updates)
+    print_graph_basic(deps_graph)
+
+    dependencies = deps_graph.nodes[1:]
+    resolved_ranges = list(deps_graph.resolved_ranges.keys())
+
+    latest_recipe = []
+
+    ConanOutput().title("Checking remotes")
+
+    # Data structure to solve the repeated dependencies in graph
+    dict_nodes = {}
+    for node in dependencies:
+        if node.name in dict_nodes:
+            dict_nodes[node.name].append(node)
+        else:
+            dict_nodes[node.name] = [node]
+
+    for node_name, node_list in dict_nodes.items():
+        ref_pattern = ListPattern(node_name, rrev=None, prev=None)
+        latest_remote_ref, found_remote, recipe_ref = None, None, None
+        for remote in remotes:
+            try:
+                remote_ref_list = conan_api.list.select(ref_pattern, package_query=None,
+                                                        remote=remote)
+                if not remote_ref_list.recipes:
+                    continue
+            except Exception:
+                ConanOutput(f"{node_name} not found in remote {remote.name}")
+                continue
+            str_latest_ref = list(remote_ref_list.recipes.keys())[-1]
+            recipe_ref = RecipeReference.loads(f"{str_latest_ref}")
+            if latest_remote_ref is None or latest_remote_ref < recipe_ref:
+                latest_remote_ref = recipe_ref
+                found_remote = remote
+        latest_cache_ref = max(node_list, key=lambda x: x.ref)
+        if recipe_ref is None:
+            continue
+        if len(node_list) != 1 or latest_cache_ref.ref < latest_remote_ref:
+            latest_recipe.append((latest_cache_ref, node_list, latest_remote_ref, found_remote))
```

Comment:
This is still a bit too cluttered. Try to split it more:
- Drop the storage of full node, store the ``refs`` only in the first iteration, after all you are only using ``node.ref`` field. Simplify early.
- Iterate the version ranges, add them to the ``dict_nodes`` for the matching names.
- Iterate the remotes, capture the latest of every ``name`` in the same ``dict_nodes``. No need of extra variables or other data structures.
- Final step, filter. Remove the ones that doesn't meet a condition, like the latest one is not more modern than the existing